airbrakes/data_handling/apogee_predictor.py

```diff
@@ -188,12 +213,38 @@ def _prediction_loop(self) -> None:
                 self._current_altitude = data_packet.current_altitude
                 self._current_velocity = data_packet.vertical_velocity
 
-            # TODO: play around with this value
-            # Run apogee prediction every 100 data points:
             if len(self._accelerations) - last_run_length >= APOGEE_PREDICTION_FREQUENCY:
+                # We only want to keep curve fitting if the curve fit hasn't converged yet
+                if not self._has_apogee_converged():
+                    curve_coefficients = self._curve_fit()
+                    lookup_table = self._generate_prediction(curve_coefficients)
                 self._cumulative_time_differences = np.cumsum(self._time_differences)
-                params = self._curve_fit()
-                self._apogee_prediction_value.value = self._get_apogee(params)
+                # Predicts the apogee using the lookup table and linear interpolation
+                predicted_apogee = np.interp(
+                    self._current_velocity, lookup_table[0], lookup_table[1]
+                )
+                self._apogee_prediction_value.value = predicted_apogee
+                self._predicted_apogees = np.append(self._predicted_apogees, predicted_apogee)
```

Comment:
Is there a reason we are keeping an unbounded list of predicted apogees in the first place? Why not just a deque of say, last 20 predicted apogees?

Reply:
i was thinking of doing that but then I would have to write all the code to like push and pop stuff and I didn't feel like doing it yet

Reply:
with deque's you don't have to manage that, it just automatically pops the last element when it's full

Reply:
lol thats what i get for asking chat--I asked it what would happen and it said it like wouldn't add 💀